app/assets/javascripts/discourse/models/post-stream.js.es6

```diff
@@ -716,6 +716,19 @@ export default RestModel.extend({
     return resolved;
   },
 
+  triggerReadPost(postId, readersCount) {
+    const resolved = Ember.RSVP.Promise.resolve();
```

Comment:
This is a little confusing. Why does it create a resolved promise only to call `then` on it and not do any async work?

If the method doesn't need to do anything async but needs to return a promise, why call `then` at all?

Reply:
I did this to be consistent with the current API (returning a promise). You're right about executing the code inside a `then`, I'll execute it outside the promise.